com.unity.netcode.gameobjects/Runtime/Interest/InterestNodeStatic.cs

```diff
@@ -0,0 +1,97 @@
+    public class InterestNodeStatic<TObject> : IInterestNode<TObject>
+    {
+        // these are the objects under my purview
+        private HashSet<TObject> m_ManagedObjects;
+
+        // these are the interest kernels that we will run on the objects under my purview
+        private List<Tuple<bool, IInterestKernel<TObject>>> m_InterestKernels;
+
+        // these are the result sets that correspond to each of the kernels I'll run.
+        //  they are then reduced
+        private List<HashSet<TObject>> m_ResultSets;
+
+        public InterestNodeStatic()
+        {
+            m_InterestKernels = new();
+            m_ManagedObjects = new();
+            m_ResultSets = new();
+        }
+
+        public void AddObject(TObject obj)
+        {
+            m_ManagedObjects.Add(obj);
+        }
+
+        public void RemoveObject(TObject obj)
+        {
+            m_ManagedObjects.Remove(obj);
+        }
+
+        public void QueryFor(TObject client, HashSet<TObject> results)
+        {
+            if (m_InterestKernels.Count > 0)
+            {
+                // run all the kernels.  We don't care whether they are additive or
+                //  subtractive...yet
+                for (var i = 0; i < m_InterestKernels.Count; i++)
+                {
+                    var thisKernel = m_InterestKernels[i].Item2;
+                    var theseResults = m_ResultSets[i];
+                    theseResults.Clear();
+                    foreach (var obj in m_ManagedObjects)
+                    {
+                        if (thisKernel.QueryFor(client, obj))
+                        {
+                            theseResults.Add(obj);
+                        }
+                    }
+                }
+                // reduce.  Note, order is important to support subtractive results
+                for (var i = 0; i < m_InterestKernels.Count; i++)
+                {
+                    // additive
+                    if (m_InterestKernels[i].Item1)
+                    {
+                        results.UnionWith(m_ResultSets[i]);
+                    }
+                    // subtractive
+                    else
+                    {
+                        results.ExceptWith(m_ResultSets[i]);
+                    }
+                }
+            }
+            else
+            {
+                results.UnionWith(m_ManagedObjects);
+            }
+        }
+
+        public void AddAdditiveKernel(IInterestKernel<TObject> kernel)
+        {
+            m_ResultSets.Add(new HashSet<TObject>());
+            m_InterestKernels.Add(new Tuple<bool, IInterestKernel<TObject>>(true, kernel));
+        }
+
+        public void AddSubtractiveKernel(IInterestKernel<TObject> kernel)
+        {
+            m_ResultSets.Add(new HashSet<TObject>());
+            m_InterestKernels.Add(new Tuple<bool, IInterestKernel<TObject>>(false, kernel));
+        }
+
+        public void UpdateObject(TObject obj)
```

Comment:
What is this for?

Reply:
This is used more often in the non-static nodes, but the idea is that if you do your own storage scheme (e.g. bucket the objects by grid) but now you want to recompute which bucket that object should go in, you call this with the object.

This is in comparison to for instance having to check every frame what grid cell the object should be in.